### DISCUSSION TITLE
Android build failing?

### DISCUSSION THREAD

**Discussion by macktns** [GodotNuts/GodotFirebase#155] [Q&A] [answered]
Hey all. I'm using this addon succesfully error free on PC. For some reason after adding this addon godot fails to build a runnable android version about 4/5 times. It builds and runs correct 100% of the time on pc and I've never had this problem on android before. When it does build and run on android, it has no problems after that. Has anybody experienced this? I'm on 3.2.4RC3.  I'm not really sure how to debug this problem, but I will try anything and report back. If I use my resource loader it returns an error before finishing. If I load my scene directly it just goes to a black screen after the splash screen.

**Comment by BearDooks** [comment, marked as answer]
That is very odd indeed. @macktns is there a need for RC3? Does it work better if you use a non RC version of godot?

I have not tried exporting anything android with it yet, but I would assume it should work fine if it works on PC. 

If possible can you run logcat on the android build so we can see what is going on there?

Chuck

**Reply by macktns** [reply]
OK I got some remote debugging working. It is stuck cyclic loading the addon. I am not doing anything to load it other than the tool created autoload singleton.  It is also saying I haven't imported a bunch of resources in the addon even though I have. Like I said, it only happens some of the time and only on android. Both of these errors repeat a bunch. I could try to run it in the stable. 

E 0:00:00.358   load: Resource: 'res://addons/godot-firebase/firebase/firebase.gd' is already being loaded. Cyclic reference?
  <C++ Error>   Condition "!success" is true. Returned: RES()
  <C++ Source>  core/io/resource_loader.cpp:349 @ load()

E 0:00:00.359   _load: Failed loading resource: res://addons/godot-firebase/firestore/firestore_collection.gdc. Make sure resources have been imported by opening the project in the editor at least once.
  <C++ Error>   Condition "found" is true. Returned: RES()
  <C++ Source>  core/io/resource_loader.cpp:279 @ _load()

**Reply by macktns** [reply]
Well 3.2.3 works perfect. 3.2.4 brings the android app bundle and other android bug fixes,.

**Reply by BearDooks** [reply]
That is odd. I wonder if it is something that will be fixed in the final release. 

Do you have the latest version of the plugin from us as well? I know we just made some changes within the last few days to fix some issues.

**Reply by BearDooks** [reply]
@macktns Just checking in to see if you had a chance to look into anything else.

**Reply by macktns** [reply]
Hello I am sorry for not getting back sooner! The RC4 actually fixed this problem! I cannot find the report now, but I believe 3.2.4 introduced a cyclical dependency error that was corrected in RC4. Apologies for not testing on stable!
